### DISCUSSION TITLE
Allow to fail if some third party action has not occurred

### DISCUSSION THREAD

**Discussion by sunpietro** [cypress-io/cypress#16027] [Questions and Help] [answered]
Hi all, I have a situation where we're using a third-party provider for cookies notice. We're not able to control it and what I want to test is:
1. if a cookie popup appears, then close it,
2. otherwise, skip it (allow to fail).

How can I do something like that in Cypress tests?

**Comment by bahmutov** [comment, marked as answer]
This is what we call conditional testing and is really not recommended https://on.cypress.io/conditional-testing

Still, you can work around it, see https://glebbahmutov.com/cypress-examples/7.1.0/recipes/conditional-testing.html

**Reply by sunpietro** [reply]
Thank you. I'm aware of all the downsides the conditional testing has, but this is a piece of software we're out of control.
